src/s3.ts

```diff
@@ -0,0 +1,39 @@
+import * as path from "node:path";
+import * as fs from "node:fs/promises";
+import * as zlib from "node:zlib";
+import { output_dir_path } from "./util.js";
+import {
+  CSV_FILENAME,
+  GZIP_CSV_FILENAME,
+  OUTPUT_DIR,
+  S3_BUCKET,
+} from "./constants.js";
+import { Event } from "./types.js";
+import { S3Client, PutObjectCommand, PutObjectCommandInput } from "@aws-sdk/client-s3";
+
+const client = new S3Client(config);
+
+export const writeToS3 = async (event: Event) => {
```

Comment:
Taking in an event as an argument suggests we're able to write individual events, which isn't really the case. What if, instead, this took in an on-disk path?